### PR TITLE
Account for new ledger refactor.

### PR DESCRIPTION
**Pull request by lehins**
Recent PR to ledger: input-output-hk/cardano-ledger#2901 introduced
a lot of breaking changes. This PR accounts for those changes.

# Description

_description of the pull request, if it fixes a particular issue it should link
the PR to a particular issue, see
[ref](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)_

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

**Comment by JaredCorduan** [comment]
The fact that this PR, an integration of a massive refactor in the ledger, is only ~100 lines, is such a clear indication of what a spectacular job @edsko and @mrBliss did to abstract the ledger inside of consensus.

File: ouroboros-consensus-shelley/ouroboros-consensus-shelley.cabal [hunk 1]
```diff
@@ -69,6 +69,7 @@ library
                      , deepseq
                      , measures
                      , mtl               >=2.2   && <2.3
+                     , microlens
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/ouroboros-consensus-shelley.cabal:72]: microslens < mtl by alphabet :)

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs [hunk 2]
```diff
@@ -133,7 +126,10 @@ type EraCrypto era = Crypto era
 -- needed to determine the hard fork point. In the future this should be
 -- replaced with an appropriate API - see
 -- https://github.com/input-output-hk/ouroboros-network/issues/2890
-class ( SL.ShelleyBasedEra era
+class ( Core.EraSegWits era
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:129]: The Haddock on this class is now out of date; eg it mentions `SL.ShelleyBasedEra`. Do you feel able to update it?

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs [hunk 3]
```diff
@@ -133,7 +126,10 @@ type EraCrypto era = Crypto era
...
+      , SL.ApplyTx era
+      , SL.ApplyBlock era
+      , SL.CanStartFromGenesis era
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:132]: We confirmed on a call with Ledger that their refactoring replaced `ShelleyBasedEra` by these constraints.

**Review by nfrisby — APPROVED** [review]
One action requested (updating the Haddock). Otherwise, Approved. Thanks!

File: ouroboros-consensus-shelley/ouroboros-consensus-shelley.cabal (hunk 1 above)
Comment (lehins) [reply ouroboros-consensus-shelley/ouroboros-consensus-shelley.cabal:71]: I can't sort by more than the first letter in my mind, I need a program for that :stuck_out_tongue_winking_eye: 

Fixed

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs (hunk 2 above)
Comment (lehins) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:127]: I've updated the haddock.

**Comment by lehins** [comment]
bors merge

**Pull request merged**